PowerPointLabs/PowerPointLabs/ELearningLab/Views/AudioSettingsViews/AudioMainSettingsPage.xaml.cs

```diff
@@ -170,6 +195,58 @@ private void PreviewCheckBox_Unchecked(object sender, RoutedEventArgs e)
             AudioSettingService.IsPreviewEnabled = false;
         }
 
+        private void EditRankingButton_Clicked(object sender, RoutedEventArgs e)
+        {
+            rankedAudioListCache = AudioSettingService.preferredVoices.Select(x => (IVoice)x.Clone()).ToList();
+            editPreferenceButton.Visibility = Visibility.Collapsed;
+            audioListView.Visibility = Visibility.Visible;
+            updatePreferenceButton.Visibility = Visibility.Visible;
+            preferredAudioListView.Visibility = Visibility.Collapsed;
+        }
+
+        private void RankingComboBox_SelectionChanged(object sender, RoutedEventArgs e)
+        {
+            ComboBox comboBox = sender as ComboBox;
+            if (comboBox.SelectedIndex > 0)
+            {
+                int rank = comboBox.SelectedIndex;
+                if (rankToComboBoxMapping.ContainsKey(rank))
+                {
+                    ComboBox previousComboBox = rankToComboBoxMapping[rank];
+                    if (!comboBox.Equals(previousComboBox))
+                    {
```

Comment:
Would it be better to add some comments here for the nested if statements? It is hard to understand the code here...

Reply:
Comments added!

Reply:
This method is trying to handle the case when item A has ranking 1, and subsequently user changes item B to have ranking 1. In that case, the ranking for item A should be removed (set to empty).